src/Pulsar.Client/Internal/ProducerImpl.fs

```diff
@@ -228,8 +262,9 @@ type internal ProducerImpl<'T> private (producerConfig: ProducerConfiguration, c
                 stats.IncrementSendFailed()
                 tcs.SetException(ex))
         else
+            let encryptBatchPayload = encrypt(metadata, encodedBatchPayload)
```

Comment:
encryptedBatchPayload, also please rename encodedBatchPayload to compressedBatchPayload